tests/unit/schema/test_api.py

```diff
@@ -2602,3 +2607,374 @@ def test_has_iam_mixin():
     })
     api_schema = api.API.build(fd, 'google.example.v1', opts=opts)
     assert api_schema.has_iam_mixin
+
+
+def get_file_descriptor_proto_for_method_settings_tests(
```

Comment:
Thanks for factoring this out.  It's much easier to follow!


Minor, non-blocking: I thought I left the following comment when I reviewed last week, but I don't see it on GitHub so it must not have gotten saved. I'll put it here just for your consideration. It's less pressing now that you've done this refactoring, but it would make it nicer when defining the various fields, etc.

If it's trivial to read from a proto "file" (string) in tests, it might be simpler to simply encode the protofile inline rather than calling `make_file_pb2` and it's ilk... something like
```
proto_descriptor = make_protodescriptor("""
package xx;

message Foo { 
  string foo = 1;
}
""")
```

(This is similar in spirit to what you do with the YAML file below!)

This would make it easier to keep the proto concepts being tested next to the code without a lot of boilerplate. If it's not trivial to do now, let's defer it. (But I personally would love to eventually have most of our tests use this declarative style rather than the harder-to-follow procedural set-up of they test cases.)

Reply:
I created https://github.com/googleapis/gapic-generator-python/issues/1986 as a follow up issue to explore whether this is feasible